liquid-staking/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+#![no_std]
+
+use multiversx_sc::imports::*;
+
+pub mod common;
+pub mod delegation;
+pub mod liquidity_pools;
+
+#[multiversx_sc::contract]
+pub trait LiquidStaking:
+    liquidity_pools::LiquidityPoolModule
+    + delegation::DelegationModule
+    + common::storage::CommonStorageModule
+{
+    #[init]
+    fn init(&self, delegation_address: ManagedAddress) {
+        require!(
```

Comment:
better extract this into a function / endpoint `add_delegation_contract`. In the future you might have this contract working with more than 1 delegation contract.

Reply:
alright, in this case I will make the storage mapper to a `UnorderedSetMapper`

Reply:
Nevermind, this won't work, I created a `SingleValueMapper` with an index as `contract_name`